### PR TITLE
[luci-micro] Introduce luci-micro

### PR DESCRIPTION
**Pull request by SlavikMIPT**
This commit introduces luci-interpreter build for microcontrollers.

ONE-DCO-1.0-Signed-off-by: Vyacheslav Bazhenov <slavikmipt@gmail.com>

### CODE REVIEW

**Comment by jinevening** [comment]
It would be better to link libraries rather than copy all library source codes.

**Comment by SlavikMIPT** [comment]
> It would be better to link libraries rather than copy all library source codes.

I am working on this

File: compiler/luci-micro/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+.mbedbuild
+cmake_build/
```
Comment (binarman) [inline compiler/luci-micro/.gitignore:2]: Add end of line here:
```suggestion
cmake_build/

```

File: compiler/luci-micro/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,116 @@
+# Copyright (c) 2021 ARM Limited. All rights reserved.
```
Comment (binarman) [inline compiler/luci-micro/CMakeLists.txt:1]: We don't add license plates in cmake files, is it from mbed-os code?

File: compiler/luci-micro/import_libs.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+# 	with open(filename, 'r') as f:
+# 	   lines = f.readlines()
+# 	for line in lines:
+# 	    match = re.search(pattern, line)
+# 	    if not match:
+# 	        new_file.append(line)
+
+# 	with open(filename, 'w') as f:
+# 	     f.seek(0)
+# 	     f.writelines(new_file)
```
Comment (binarman) [inline compiler/luci-micro/import_libs.py:32]: end of line
```suggestion
# 	     f.writelines(new_file)

```

File: compiler/luci-micro/import_libs.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+shutil.copytree('../pepper-str', './lib/pepper-str', dirs_exist_ok=True)
+shutil.copytree('../logo', './lib/logo', dirs_exist_ok=True)
+shutil.copytree('../logo-core', './lib/logo-core', dirs_exist_ok=True)
+shutil.copytree('../loco', './lib/loco', dirs_exist_ok=True)
+shutil.copytree('../locomotiv', './lib/locomotiv', dirs_exist_ok=True)
+shutil.copytree('../angkor', './lib/angkor', dirs_exist_ok=True)
+shutil.copytree('../luci/lang', './lib/luci/lang', dirs_exist_ok=True)
+shutil.copytree('../luci/import', './lib/luci/import', dirs_exist_ok=True)
+shutil.copytree('../luci/env', './lib/luci/env', dirs_exist_ok=True)
+# shutil.copytree('../luci/log', './lib/luci/log', dirs_exist_ok=True)
```
Comment (binarman) [inline compiler/luci-micro/import_libs.py:15]: Do I understand it right, that you copy luci-interpreter (and all dependencies) in this directory, and do not modify original code?

**Review by binarman — COMMENTED** [review]
What is the problem with nncc_common component?
I see you disable it everywhere and change a lot of files.
Maybe it is easier to modify this component inself, so everything will be in one place?

**Pull request closed without merging**